inst/extdata/dataset_title_length-check.xml

```diff
@@ -43,6 +43,14 @@ mdq_result <- list(status=result$status, output=list(list(value=result$output)))
    </namespaces>
    <namespaceAware>true</namespaceAware>
   </selector>
+  <selector>
```

Comment:
usually when a dialect is added, the xpath is added to the selector element like this 
```
<selector>
    <name>datasetTitle</name>
    <xpath>/resource/titles/title |
             /*/title |
             /eml/dataset/title |
             /*/identificationInfo/*/citation/CI_Citation/title
      </xpath>
  </selector>
```

but I'm not sure how we want to handle this with the jq queries. we could either shoehorn it into the existing structure or add an element to the schema. I'll need to think about this one

Reply:
The syntax above is a valid XPath expression. I don't think mixing a json-path expression in would make sense -- I feel that we need abother field for the json-path expression. Or maybe we need to add a more extensible query block that is repeatable (e..g, `<query syntax="json-path">.name</query>` or something like that. And maybe it could (or should?) include a dialect attribute that says which metadata dialect these apply to (compared to what we did before to cram all of the selctors into a single xpath expression).

Reply:
@mbjones I also posted about this in the DataONE slack and agree that mixing expressions doesn't make sense. I like the idea of the repeatable query, especially when combined with the dialect, but that is a pretty major refactor and we definitely have to make sure we are backwards compatibile

Reply:
Yeah, fully agree we should be careful here. At a minimum we could add the new element and make it optional and repeatable, which hopefully doesn't in ay way interfere with or change how the `xpath` element is used. It might not be too bad to refactor it all though, given that you'll need to introduce new logic in the Java code to handle json expressions differently from xpath. so seems like you'll need to refactor that part of the code anyways. A nice transition approach would be to 1) make `xpath` and `query` optional; 2) modify your selector processing code to extract all xpaths and json paths from both fields, and then 3) launch the queries needed to get and combine the values to pass on to the check code.